src/parser/core/modules/CooldownDowntime.tsx

```diff
@@ -0,0 +1,264 @@
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import {ActionLink} from 'components/ui/DbLink'
+import {getDataBy} from 'data'
+import ACTIONS from 'data/ACTIONS'
+import {Action} from 'data/ACTIONS/ACTIONS'
+import {CastEvent} from 'fflogs'
+import Module, {dependency} from 'parser/core/Module'
+import {Requirement, Rule} from 'parser/core/modules/Checklist'
+import React from 'react'
+import Checklist from './Checklist'
+import Downtime from './Downtime'
+
+const DEBUG_EXPECTED_CALCULATION = false && process.env.NODE_ENV !== 'production'
+
+interface CooldownReset {
+	/**
+	 * One or more skills that trigger a cooldown reset.
+	 */
+	actions: Action[],
+	/**
+	 * The amount of time in ms that an action deducts from the remaining
+	 * cooldown of the affected cooldowns.
+	 */
+	refundAmount?: number,
+	fullReset?: boolean
```

Comment:
Yeah, Kelos's original approach was to just use refundAmount and have full resets be the full cooldown as well... I went with the separate code path because I thought it was a little more clear to someone consuming CooldownDowntime, and because using the full cooldown length results in "wasted reset time" being incremented by some of the full reset time, which doesn't usually make sense for skills like Acceleration resetting RDM's move blocks.  Of course, we're not really doing anything sexy with wasted reset time yet, just counting it, so I'm fine with going to a single refundAmount on the interface.

Reply:
For now, I'm going to go ahead and go back to just using `refundAmount`, as no current consumers of CooldownDowntime have fullReset resetters anyway, and we're not actually doing anything with the wastage we're tracking.  If it becomes a future issue, we can tackle it then.